Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp

```diff
@@ -959,6 +959,54 @@ namespace Tensile
                 }
             };
 
+            struct GlobalSplitUCheckK
+                : public Predicate_CRTP<GlobalSplitUCheckK, ContractionProblem>
+            {
+                enum
+                {
+                    HasIndex = false,
+                    HasValue = true
+                };
+                GlobalSplitUCheckKPacket value;
```

Comment:
There's no reason the two values here couldn't just be members of this class, we would just need a custom serialization traits class for this predicate.

Reply:
hi @sdquiring ,

Current Property in Tensile/Properties.py only accept one index and one value.
If we want to pass two values to new client predictor, we have to pack it to one value.

So how to  achieve " custom serialization traits class for this predicate" ??

Reply:
> custom serialization traits class

This means implementing the serialization code manually instead of simply inheriting from `AutoMappingTraits` in ContractionPredicates.hpp.  You just need to implement:

```cpp
static void mapping(IO& io, Object& obj)
```

You would also need to ensure that your properties get written out by including them in the dictionary returned by `state` in the Property base class in Python:

```python
def state(self):
```

Reply:
Hi @sdquiring ,

I modify Properties.py to add one more value parameter and extend AutoMapping to accept 2 values if needed.
That is similar to your suggestion and should be more flexible than customize case.
How do you think about this change??

Thanks,
Tony.

Reply:
Sorry, this slipped past me in my inbox.

No, I don't like this solution.  If there are two values, what is the meaning of each variable? It can't be named something meaningful, just `value` and `value2`.

Reply:
ok, we discard modification for check K in each GSU work.
Only keep workspace limitation for Tensile training.
How about that??

Reply:
Your solution to "do this a different way" is to not do it at all?

Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp
```cpp
struct GlobalSplitUCheckK: public Predicate_CRTP<GlobalSplitUCheckK, ContractionProblem>
{
    enum
    {
        HasIndex = true,
        HasValue = true
    };
    size_t index;
    size_t value;
    size_t k;
    // ... Need to implement manual toString() in addition to Type() and operator()().

}
```

Tensile/Source/lib/include/Tensile/Serialization/ContractionPredicates.hpp
```cpp
template <typename IO>
struct MappingTraits<Predicates::Contraction::GlobalSplitUCheckK, IO>
{
    using Predicate = Predicates::Contraction::GlobalSplitUCheckK;
    using iot     = IOTraits<IO>;

    static void mapping(IO& io, Predicate& pred)
    {
        iot::mapRequired(io, "index", pred.index);
        iot::mapRequired(io, "value", pred.value);
        iot::mapRequired(io, "k", pred.k);
    }
};

```